fMRIVolume/scripts/OneStepResampling.sh

```diff
@@ -209,23 +257,29 @@ ${FSLDIR}/bin/fslmerge -tr ${OutputfMRI}_mask $FrameMergeSTRINGII $TR_vol
 
-# Combine transformations: gradient non-linearity distortion + fMRI_dc to standard
-${FSLDIR}/bin/convertwarp --relout --rel --ref=${WD}/${T1wImageFile}.${FinalfMRIResolution} --warp1=${GradientDistortionField} --warp2=${OutputTransform} --out=${WD}/Scout_gdc_MNI_warp.nii.gz
-${FSLDIR}/bin/applywarp --rel --interp=spline --in=${ScoutInput} -w ${WD}/Scout_gdc_MNI_warp.nii.gz -r ${WD}/${T1wImageFile}.${FinalfMRIResolution} -o ${ScoutOutput}
+if [ ${fMRIReferencePath} = "NONE" ] ; then
+  # Combine transformations: gradient non-linearity distortion + fMRI_dc to standard
+  ${FSLDIR}/bin/convertwarp --relout --rel --ref=${WD}/${T1wImageFile}.${FinalfMRIResolution} --warp1=${GradientDistortionField} --warp2=${OutputTransform} --out=${WD}/Scout_gdc_MNI_warp.nii.gz
+  ${FSLDIR}/bin/applywarp --rel --interp=spline --in=${ScoutInput} -w ${WD}/Scout_gdc_MNI_warp.nii.gz -r ${WD}/${T1wImageFile}.${FinalfMRIResolution} -o ${ScoutOutput}
+fi
 
 # Create spline interpolated version of Jacobian  (T1w space, fMRI resolution)
 #${FSLDIR}/bin/applywarp --rel --interp=spline -i ${JacobianIn} -r ${WD}/${T1wImageFile}.${FinalfMRIResolution} -w ${StructuralToStandard} -o ${JacobianOut}
-#fMRIToStructuralInput is from gdc space to T1w space, ie, only fieldmap-based distortions (like topup)
+#fMRI2Struct is from gdc space to T1w space (optionally through an external reference), ie, only fieldmap-based distortions (like topup)
 #output jacobian is both gdc and topup/fieldmap jacobian, but not the to MNI jacobian
 #JacobianIn was removed from inputs, now we just compute it from the combined warpfield of gdc and dc (NOT MNI)
 #compute combined warpfield, but don't use jacobian output because it has 8 frames for no apparent reason
 #NOTE: convertwarp always requires -o anyway
-${FSLDIR}/bin/convertwarp --relout --rel --ref=${fMRIToStructuralInput} --warp1=${GradientDistortionField} --warp2=${fMRIToStructuralInput} -o ${WD}/gdc_dc_warp --jacobian=${WD}/gdc_dc_jacobian
+if [ "$fMRIReferenceReg" == "nonlinear" ]; then
+  ${FSLDIR}/bin/convertwarp --relout --rel --ref=${WD}/postmc2struct_warp --warp1=${GradientDistortionField} --warp2=${WD}/postmc2struct_warp -o ${WD}/gdc_dc_warp --jacobian=${WD}/gdc_dc_jacobian
+else
+  ${FSLDIR}/bin/convertwarp --relout --rel --ref=${fMRIToStructuralInput} --warp1=${GradientDistortionField} --warp2=${fMRIToStructuralInput} -o ${WD}/gdc_dc_warp --jacobian=${WD}/gdc_dc_jacobian
+fi
 #but, convertwarp's jacobian is 8 frames - each combination of one-sided differences, so average them
 ${FSLDIR}/bin/fslmaths ${WD}/gdc_dc_jacobian -Tmean ${WD}/gdc_dc_jacobian
 
 #and resample it to MNI space
-${FSLDIR}/bin/applywarp --rel --interp=spline -i ${WD}/gdc_dc_jacobian -r ${WD}/${T1wImageFile}.${FinalfMRIResolution} -w ${StructuralToStandard} -o ${JacobianOut}
+${FSLDIR}/bin/applywarp --rel --interp=trilinear -i ${WD}/gdc_dc_jacobian -r ${WD}/${T1wImageFile}.${FinalfMRIResolution} -w ${StructuralToStandard} -o ${JacobianOut}
```

Comment:
Is the change from spline to trilinear interpolation intentional here?

Reply:
As I recall, there were ringing artifacts in the jacobian when using spline resampling, so this change was intentional.

Reply:
Right

Reply:
Right.  @sivcek Please add a comment explaining why it is set to trilinear, possibly even including the URL to the GitHub discussion of the issue.  Thx.

Reply:
I added:
```
#and resample it to MNI space
# Note that trilinear instead of spline interpolation is used with the purpose to minimize the ringing artefacts that occur 
# with downsampling of the jacobian field and are then propagated to the BOLD image itself.
```
The discussion occured over email and can not be linked to.